articles/using_synapse_as_an_oauth_server.md

```diff
@@ -0,0 +1,129 @@
+
+print(clientIdAndSecret)
+
+```
+
+The client URI, policy URI, and terms of service URI are optional, as is the `userinfo_signed_response_alg` parameter.  Further, you can optionally include a `sector_identifier_uri` parameter.  This is an advanced feature described [here](https://openid.net/specs/openid-connect-registration-1_0.html#SectorIdentifierValidation) and is relevant if the client uses multiple hosts because Synapse only returns `pairwise` subject values to its OAuth clients.
+
+
+The returned `client_id` and `client_secret` will be needed later when getting an access token.  The secret is only returned once.  (Synapse does not keep it.)  If lost, you can generate a new secret but the previous one will be invalidated.
+
+You can retrieve, update, and delete your client programmatically as well:
+
+```python
+clientMetaData = syn.restGET(uri='/oauth2/client/'+client_id, endpoint=syn.authEndpoint)
+
+clientMetaData['policy_uri'] = 'https://yourhost.com/updated_policy'
+
+clientMetaData = syn.restPUT(uri='/oauth2/client/'+client_id, endpoint=syn.authEndpoint, clientMetaData)
+
+syn.restDELETE(uri='/oauth2/client/'+client_id, endpoint=syn.authEndpoint)
+
+```
+
+To login via Synapse your client should redirect the browser from your application to `https://signin.synapse.org`, with the standard OAuth 2.0 request parameters:
+
+- `client_id`=`<your client id>`
+- `scope`=`openid`
+- `redirect_uri`=`<the redirect uri registered with your client>`
+- `response_type`=`code`
+- `state`=`<any state you want returned>`
+- `nonce`=`<some string to be returned in the ID token>`
+- `claims`=`<a JSON object>`
+
+Synapse supports the `claims` request parameter, a JSON document containing the details of the user identity information you would like returned, as described [here](https://openid.net/specs/openid-connect-core-1_0.html#ClaimsParameter).  The list of supported claims is given here TBD.  For most claims the value to include in the JSON document is `null`.  The exception is the `team` claim, for which you provide the IDs of one or more teams, the membership of which you wish to inquire about.  Synapse will return the IDs of the subset of the given list of teams to which the user belongs.  Here is an example of a claims parameter JSON object:
+
+```
+ {
+   "id_token":
+    {
+     "given_name": null,
+     "family_name": null,
+     "is_certified": null,
+     "team": {"values":["273957", "3385814"]}
+    },
+   "userinfo":
+    {
+     "given_name": null,
+     "family_name": null,
+     "is_certified": null,
+     "team": {"values":["273957", "3385814"]}
+    }
+  }
+```
+
+Following the user's consent, Synapse will redirect back to your specified redirect URI with an authorization code and your 'state' data.  You can then exchange the authorization code for a ID token and access token in the standard way.  Here is an example using cUrl:
```

Comment:
I don't understand the word `consent` here.

Reply:
The user has to click "Allow" in the web browser for Synapse to send the info to the linked client application.